modules/nf-core/prokka/main.nf

```diff
@@ -31,18 +31,26 @@ process PROKKA {
     task.ext.when == null || task.ext.when
 
     script:
-    def args = task.ext.args   ?: ''
-    prefix   = task.ext.prefix ?: "${meta.id}"
-    def proteins_opt = proteins ? "--proteins ${proteins[0]}" : ""
-    def prodigal_tf = prodigal_tf ? "--prodigaltf ${prodigal_tf[0]}" : ""
+    def args             = task.ext.args   ?: ''
+    prefix               = task.ext.prefix ?: "${meta.id}"
+    def fasta_compressed = fasta.getExtension() == "gz" ? true : false
+    def input            = fasta_compressed ? fasta.toString() - ~/\.gz$/ : fasta
```

Comment:
I guess this could be simplified  too, because if the filename doesn't end with `.gz` the pattern should not remove anything anyway.